### PR TITLE
Update AKS Run Command

### PR DESCRIPTION
**Pull request by dciborow**
## Description

The following changes are required so that this module can be used with a Marketplace application within our new Unreal Cloud DDC module - https://github.com/Azure/bicep-registry-modules/pull/276

This introduces a **BREAKING_CHANGE** by requiring the location parameter be set, as required by Azure Marketplace per https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/linter-rule-explicit-values-for-loc-params

Aligning newOrExisting with the createUI per https://learn.microsoft.com/en-us/azure/azure-resource-manager/managed-applications/microsoft-common-serviceprincipalselector

## Updating an existing module

- [x] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.
- [x] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [x] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [x] I have updated the examples in README with the latest module version number.

### CODE REVIEW

File: modules/deployment-scripts/aks-run-command/main.bicep [hunk 1]
```diff
@@ -31,35 +32,39 @@ param commands array
...
-  'OnSuccess'
-  'OnExpiration'
-  'Always'
-])
+@allowed([ 'OnSuccess', 'OnExpiration', 'Always' ])
 @description('When the script resource is cleaned up')
 param cleanupPreference string = 'OnSuccess'
 
-resource aks 'Microsoft.ContainerService/managedClusters@2022-09-01' existing = {
+@description('Set to true when deploying template across tenants')
```
Comment (Gordonby) [inline modules/deployment-scripts/aks-run-command/main.bicep:39]: Can you expand on this scenario description further?

**Review by Gordonby — APPROVED** [review]
Looks like a good update.

File: modules/deployment-scripts/aks-run-command/main.bicep (hunk 1 above)
Comment (dciborow) [reply modules/deployment-scripts/aks-run-command/main.bicep:39]: When going cross-tenant, within a Marketplace application, without this the Marketplace's SP tries to run the custom script step and we can't RBAC failures.

We have tried without but have found that it is needed.

**Pull request merged**